activeviam/activeui-migration
# UI-7946 - CLI requires from-version and to-version arguments

## Description

With this PR, the CLI now requires to new arguments: `from-version` (alias `f`), and `to-version`(alias `t`). The users hence now need to specify the versions they want to migrate their content from and to.

## Testing procedure

- Pull this branch and build the repo
- Outside of this repo, create a `migration-test` folder
- Init this folder (`npm init`) and install you local `activeui-migration` dependency using [`npm link`](https://www.geeksforgeeks.org/how-to-install-a-local-module-using-npm/)
- Follow the steps [here](https://github.com/activeviam/activeui-migration#installation) to finish the setup
- Run the migration command line adding the `-f` and `-t` arguments:
  - `-f 5.0 -t 5.1` should invalidate the `-f` argument
  - `-f 4.3 -t 5.1` should invalidate the `-t` argument
- if you want to test the case where both arguments are valid but don't reflect a valid combination (e.g. `-f 5.0 -t 4.3`), you would need to add other versions in the `migrationFunctions` array in `bin.ts`
- finally, the current default case `-f 4.3 -t 5.0` should still work
